article-store/events/utils.py

```diff
@@ -0,0 +1,122 @@
+from contextlib import contextmanager
+from functools import partial, wraps
+import json
+from time import gmtime, strftime
+from typing import (
+    Any,
+    Callable,
+    ContextManager,
+    Dict
+)
+import uuid
+
+import pika
+from pika.exceptions import ConnectionClosed
+from pika.adapters.blocking_connection import BlockingChannel
+
+from .conf import (
+    ARTICLES_EXCHANGE,
+    BROKER_PARAMS,
+    DEFAULT_QUEUES
+)
+
+
+DELIVERY_MODE_PERSISTENT = 2
+
+
+def create_message(msg_type: str, identifier: str, data: Dict) -> Dict:
+    """Create a message `dict` based on a standard schema.
+
+    :return: Dict
+    """
+    return {
+        "eventId": str(uuid.uuid1()),
+        "happenedAt": strftime("%Y-%m-%dT%H:%M:%S+00:00", gmtime()),
+        "aggregate": {
+            "service": "article-store",
+            "name": "article-version",
+            "identifier": identifier,
+        },
+        "type": msg_type,
+        "data": data or {}
+    }
+
+
+@contextmanager
+def get_channel() -> ContextManager[BlockingChannel]:
+    """Handles the creation and clean up of a connection,
+    giving the caller a connection channel to use.
+    :return: class: `BlockingChannel`
+    """
+    try:
+        connection = pika.BlockingConnection(parameters=BROKER_PARAMS)
+        yield connection.channel()
+        connection.close()
+    except ConnectionClosed as err:
+        print(err)
+        raise
+
+
+def setup_exchanges_and_queues(func) -> Callable[..., None]:
```

Comment:
We don't need a queue here? Though guess this is just copied from earlier work?

Reply:
Yes, it was brought in from previous work, but also makes sure the queue(s) you are binding the exchange(s) to exists before the actual binding. This may be subject to change if we can guarantee a queues existence but for now, I think we have to do it this way.

Reply:
But in this case we don't need the `articles` queue?